examples/temporal_graph_learning/gnn_encoder.py

```diff
@@ -0,0 +1,218 @@
+from graphstorm import model as gsmodel
+
+from model_utils import get_unique_nfields
+from model_utils import to_per_field_nfeats
+from model_utils import rel_field_map
+from model_utils import rel_name_map
+from model_utils import average_over_fields
+from model_utils import get_unique_etype_triplet
+
+from model_utils import merge_multi_blocks, get_merge_canonical_etype_mapping
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from dgl.nn import GATConv
+
+class TemporalRelationalGraphEncoder(gsmodel.gnn_encoder_base.GraphConvEncoder):
+    """ Relational temporal graph conv encoder.
+
+    Parameters
+    ----------
+    g: DistGraph
+        The distributed graph
+    h_dim : int
+        Hidden dimension
+    out_dim : int
+        Output dimension
+    num_hidden_layers : int
+        Number of hidden layers. Total GNN layers is equal to num_hidden_layers + 1. Default 1
+    num_heads : int
+        Number of attention heads
+    dropout : float
+        Dropout. Default 0.
+    """
+    def __init__(self, g, h_dim, out_dim, num_hidden_layers, num_heads=0, dropout=0):
+        super(TemporalRelationalGraphEncoder, self).__init__(h_dim, out_dim, num_hidden_layers)
+        self.node_fields = get_unique_nfields(g.etypes)
+        for _ in range(num_hidden_layers):
+            self.layers.append(
+                ResidualTemporalRelationalGraphConv(
+                    g,
+                    in_feat=h_dim,
+                    out_feat=h_dim,
+                    num_heads=num_heads,
+                    dropout=dropout,
+                    activation=nn.LeakyReLU(),
+                )
+            )
+        self.layers.append(
+            ResidualTemporalRelationalGraphConv(
+                g,
+                in_feat=h_dim,
+                out_feat=out_dim,
+                num_heads=num_heads,
+                dropout=dropout,
+                activation=nn.LeakyReLU(),
+            )
+        )
+
+    def forward(self, blocks, inputs):
+        """Forward computation.
+
+        Parameters
+        ----------
+        blocks: DGL MFGs
+            Sampled subgraph in DGL MFG
+        inputs: dict[str, torch.Tensor]
+            Input node feature for each node type.
+        """
+        h = to_per_field_nfeats(inputs, self.node_fields)
```

Comment:
this assumes a format that stores the node features at different timestamps. the format needs to be explicitly explained in the readme.

Reply:
I short paragraph is added to README's **Implementation** section.